src/main/java/teammates/common/datatransfer/attributes/FeedbackResponseAttributes.java

```diff
@@ -167,40 +167,44 @@ public void sanitizeForSaving() {
         // nothing to sanitize before saving
     }
 
+    private FeedbackResponseDetails deserializeResponseFromMetaData(String responseMetaData) {
+        Class<? extends FeedbackResponseDetails> responseDetailsClass = getFeedbackResponseDetailsClass();
+
+        if (responseDetailsClass == FeedbackTextResponseDetails.class) {
+            // For Text questions, the questionText simply contains the question, not a JSON
+            // This is due to legacy data in the data store before there are multiple question types
+            return new FeedbackTextResponseDetails(responseMetaData);
+        }
+        return JsonUtils.fromJson(responseMetaData, responseDetailsClass);
+    }
+
     /**
-     * Converts the given Feedback*ResponseDetails object to JSON for storing.
+     * Converts the feedbackResponseDetails variable to JSON String for storing.
      */
-    public void setResponseDetails(FeedbackResponseDetails responseDetails) {
+    public String getSerializedFeedbackResponseDetail() {
         if (responseDetails == null) {
             // There was error extracting response data from http request
-            responseMetaData = "";
+            return "";
         } else if (responseDetails.questionType == FeedbackQuestionType.TEXT) {
             // For Text questions, the answer simply contains the response text, not a JSON
             // This is due to legacy data in the data store before there were multiple question types
-            responseMetaData = responseDetails.getAnswerString();
+            return responseDetails.getAnswerString();
         } else {
-            responseMetaData = JsonUtils.toJson(responseDetails, getFeedbackResponseDetailsClass());
+            return JsonUtils.toJson(responseDetails, getFeedbackResponseDetailsClass());
         }
     }
 
-    /**
-     * Retrieves the Feedback*ResponseDetails object for this response.
-     * @return The Feedback*ResponseDetails object representing the response's details
-     */
     public FeedbackResponseDetails getResponseDetails() {
+        return responseDetails;
+    }
 
-        if (isMissingResponse()) {
-            return null;
-        }
-
-        Class<? extends FeedbackResponseDetails> responseDetailsClass = getFeedbackResponseDetailsClass();
+    public void setResponseDetailsByMetaData(String metaData) {
```

Comment:
Why there is still a method to set the serialised metaData directly?

Reply:
This method is used in `FeedbackSubmissionEditSaveAction.java`, but you mentioned it is not used, should I remove `FeedbackSubmissionEditSaveAction.java` along with all tests?